lib/common/grpc-service.js

```diff
@@ -266,6 +266,25 @@ GrpcService.prototype.request = function(protoOpts, reqOpts, callback) {
     request: function(_, onResponse) {
       respError = null;
 
+      if (stream) {
```

Comment:
> That worries me a little bit, only because I think BigTable is often used for extremely large datasets. 

That's a great point, though I don't know if it's fair to shut down callbacks from all stream APIs just based on `getRows`. `sampleRowKeys` shouldn't produce such a large set.

We can use the docs to recommend against using a callback when worried about the result size, but we should still offer the tool. If we are still worried, we could monitor the buffer size and just close the stream after it fills to a certain amount, execute the callback with an error and the results we were able to capture... I wouldn't be surprised if there was already a library which handles something similar to that already.

> then we would have to transform a stream to a callback and then back to a stream, which feels pretty inefficient.

I'm not sure where this would fit in our API, but it seems like we should be able to just do:

```js
runStreamMethod: function(callback) {
  if (callback) {
    this.runStreamMethod.pipe(concat(callback));
    return;
  }

  return this.requestAsStream(...);
}
```